packages/gatsby/src/query/query-runner.js

```diff
@@ -128,5 +126,12 @@ module.exports = async (graphqlRunner, queryJob: QueryJob) => {
     isPage: queryJob.isPage,
   })
 
+  if (process.env.GATSBY_PAGE_BUILD_ON_DATA_CHANGES && queryJob.isPage) {
```

Comment:
Let's add comment mentioning that this is here because of easy access to already calculated `resultHash` (most of other changes are (or will be) contained in `build.js` file)